src/SFA.DAS.ProviderApprenticeshipsService.Web/Orchestrators/BulkUploader.cs

```diff
@@ -0,0 +1,165 @@
+
+            return errors;
+        }
+
+        public IEnumerable<ApprenticeshipUploadModel> CreateViewModels(HttpPostedFileBase attachment)
+        {
+            string result = new StreamReader(attachment.InputStream).ReadToEnd();
+            using (TextReader tr = new StringReader(result))
+            {
+                var csvReader = new CsvReader(tr);
+                csvReader.Configuration.HasHeaderRecord = true;
+                csvReader.Configuration.IsHeaderCaseSensitive = false;
+
+                try
+                {
+                    return csvReader.GetRecords<CsvRecord>()
+                        .ToList()
+                        .Select(MapTo);
+                }
+                catch (CsvMissingFieldException exception)
+                {
+                    var exceptionData = exception.Data["CsvHelper"];
+                    _logger.Error(
+                        exception,
+                        $"Failed to create files from bulk upload. {typeof(CsvMissingFieldException)} Data CsvHelper {exceptionData}");
+                    throw new Exception("Cannot read all file");
+                }
+                catch (Exception exception)
+                {
+                    var exceptionData = exception.Data["CsvHelper"];
+                    _logger.Error(
+                        exception,
+                        $"Failed to create files from bulk upload. Exception Data CsvHelper {exceptionData}");
+                    throw new Exception("Failed to create apprentices from file");
+                }
+            }
+        }
+
+        public virtual IEnumerable<UploadError> ValidateFields(IEnumerable<ApprenticeshipUploadModel> records, List<ITrainingProgramme> trainingProgrammes)
+        {
+            var errors = new List<UploadError>();
+
+            if (!records.Any()) return new[] { new UploadError(ApprenticeshipFileValidationText.NoRecords) };
+
+            var index = 1;
+            foreach (var record in records)
+            {
+                var viewModel = record.ApprenticeshipViewModel;
+                index++;
+
+                var viewModelValidator = new ApprenticeshipBulkUploadValidator();
+                var approvalValidator = new ApprenticeshipViewModelApproveValidator();
+                var csvRecordValidator = new CsvRecordValidator();
+
+                // Validate view model 
+                var validationResult = viewModelValidator.Validate(viewModel);
+                validationResult.Errors.ForEach(m => errors.Add(new UploadError(m.ErrorMessage, m.ErrorCode, index)));
+
+                // Validate view model for approval
+                var approvalValidationResult = approvalValidator.Validate(viewModel);
+                approvalValidationResult.Errors.ForEach(m => errors.Add(new UploadError(m.ErrorMessage, m.ErrorCode, index)));
+
+                // Validate csv record
+                var csvValidationResult = csvRecordValidator.Validate(record.CsvRecord);
```

Comment:
I'm wondering if we should map both web and bulk upload objects into a single model that goes through the same validators? Instead of there being specific validators for the csv and the web? What do you think?